compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -1446,7 +1446,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         .filter_map(|projs| {
                             if let ProjectionKind::Field(field_idx, _) = projs.first().unwrap().kind
                             {
-                                if (field_idx as usize) == i { Some(&projs[1..]) } else { None }
+                                if field_idx.index() == i { Some(&projs[1..]) } else { None }
```

Comment:
Can this use `iter_enumerated` similarly to the change above?

Reply:
Ig having an `IndexList` wouldn't be too hard, maybe 🤔

Reply:
I opened #109787 to add a slice version at least :)

Reply:
Ah, nice! I was thinking about that too 😄